cypher/cypher-docs/src/test/scala/org/neo4j/cypher/docgen/AggregatingFunctionsTest.scala

```diff
@@ -84,88 +84,96 @@ class AggregatingFunctionsTest extends DocumentingTest {
+        p("All the values are collected and returned in a single list.")
         resultTable()
       }
     }
     section("count()", "functions-count") {
       p(
-        """`count()` is used to count the number of rows.
-          |`count()` can be used in two forms -- `count(*)` which just counts the number of matching rows, and `count(<expression>)`, which counts the number of non-`null` values in `<expression>`.
+        """`count()` returns the number of values or rows, and appears in two variants:
+          |
+          |* `count(*)` returns the number of matching rows, and
+          |* `count(expr)` returns the number of non-`null` values returned by an expression.
         """.stripMargin)
-      function("count(expression)", ("expression", "An expression."))
-      section("Count nodes") {
-        p("`count(*)` can be used to count the number of nodes; for example, the number of nodes connected to one node.")
+      function("count(expression)", "An Integer.", ("expression", "An expression."))
+      considerations("`count(*)` includes rows returning `null`.", "`count(expr)` ignores `null` values.", "`count(null)` returns `0`.")
+      section("Using `count(*)` to return the number of nodes") {
+        p("`count(*)` can be used to return the number of nodes; for example, the number of nodes connected to some node `n`.")
         query("MATCH (n {name: 'A'})-->(x) RETURN labels(n), n.age, count(*)", ResultAssertions((r) => {
           r.toList should equal(List(Map("labels(n)" -> List("Person"), "n.age" -> 13L, "count(*)" -> 3L)))
         })) {
-          p("This returns the start node and the count of related nodes.")
+          p("The labels and `age` property of the start node `n` and the number of nodes related to `n` are returned.")
           resultTable()
         }
       }
-      section("Group count relationship types") {
-        p("To count the groups of relationship types, return the types and count them with `count(*)`.")
+      section("Using `count(*)` to group and count relationship types") {
+        p("`count(*)` can be used to group relationship types and return the number of each of these.")
         query("MATCH (n {name: 'A'})-[r]->() RETURN type(r), count(*)", ResultAssertions((r) => {
           r.toList should equal(List(Map("type(r)" -> "KNOWS", "count(*)" -> 3L)))
         })) {
-          p("The relationship types and their group count is returned.")
+          p("The relationship types and their group count are returned.")
           resultTable()
         }
       }
-      section("Count entities") {
-        p("Instead of counting the number of results with `count(*)`, it might be more expressive to include the name of the variable you care about.")
+      section("Using `count(expression)` to return the number of values") {
+        p("Instead of simply returning the number of rows with `count(*)`, it may be more useful to return the actual number of values returned by an expression.")
```

Comment:
Changing 'variable' to 'expression' is not wrong, but I think moves away from the specific example being described.